scopesim/effects/spectral_trace_list.py

```diff
@@ -299,42 +300,61 @@ def rectify_traces(self, hdulist, xi_min=None, xi_max=None, interps=None,
         #pdu.header['FILTER'] = from_currsys("!OBS.filter_name_fw1")
         outhdul = fits.HDUList([pdu])
 
-        for i, trace_id in enumerate(self.spectral_traces):
+        for i, trace_id in enumerate(self.spectral_traces, start=1):
             hdu = self[trace_id].rectify(hdulist,
                                          interps=interps,
                                          bin_width=bin_width,
                                          xi_min=xi_min, xi_max=xi_max,
                                          wave_min=wave_min, wave_max=wave_max)
             if hdu is not None:   # ..todo: rectify does not do that yet
                 outhdul.append(hdu)
-                outhdul[0].header[f"EXTNAME{i+1}"] = trace_id
+                outhdul[0].header[f"EXTNAME{i}"] = trace_id
```

Comment:
`i` is in the range `[0, len(spec traces)]` from the enumerate call in line 302
The primary header `EXTNAME0` will (should) never contain trace data.
Trace data should only be held in the additional FITS extensions 1..N, i.e. `EXTNAME[1 ... len(spec traces)]`

Reply:
If you look a few lines further up, I also changed the enumerate to `enumerate(self.spectral_traces, start=1)`, which means `i` is in `range(1, len(self.spectral_traces) + 1)`. This (in general, not only in this example) has the benefits of more clearly marking the purpose of `i` (we want values from 1, not 0) and saving to type `+1` every time (which may cause bugs if forgotten once). Of course, this only makes sense if `i` is **only** used as `i+1` within that particular scope (which is the case here).

Reply:
Today I learned ... :)